components/nexusoperations/workflow/commands.go

```diff
@@ -220,7 +220,11 @@ func (ch *commandHandler) HandleCancelCommand(
 	if err != nil {
 		return err
 	}
-	if !nexusoperations.TransitionCanceled.Possible(op) {
+	// The operation is already in a terminal state and the terminal NexusOperation event has not just been buffered.
+	// We allow the workflow to request canceling an operation that has just completed while a workflow task is in
+	// flight since it cannot know about the state of the operation.
+	// TODO(bergundy): When we support state machine deletion, this condition will have to change.
+	if !nexusoperations.TransitionCanceled.Possible(op) && !ms.HasAnyBufferedEvent(makeNexusOperationTerminalEventFilter(attrs.ScheduledEventId)) {
```

Comment:
Shouldn't this condition be like `if !TransitionCanceled.Possible || ms.HasAnyBufferedEvent`?

Reply:
No the error is only if there isn't currently a buffered event, otherwise we ignore the command.

Reply:
So, if you are not returning an error here, then it will try to change the state to canceled in L244. And I see in your tests that no error is being returned, so it will change the state from, say, completed to canceled?

Reply:
No but it will create the cancelation state machine, there's no canceled state on the operation. It's fine to create the machine, it won't do anything if the operation is already closed.

Reply:
It's even better if we do that because the cancel requested event is applied before the completion event during replication due to delayed application of buffered events. This way, both clusters converge at the same state.